hazelcast/src/main/java/com/hazelcast/internal/config/AbstractDomConfigProcessor.java

```diff
@@ -312,33 +315,63 @@ protected void fillNativeMemoryConfig(Node node, NativeMemoryConfig nativeMemory
+                pmemConfig.addDirectoryConfig(new PersistentMemoryDirectoryConfig(getTextContent(n).trim()));
+                // we enable the persistent memory configuration for legacy reasons
+                pmemConfig.setEnabled(true);
             } else if (matches("persistent-memory", nodeName)) {
                 handlePersistentMemoryConfig(nativeMemoryConfig.getPersistentMemoryConfig(), n);
             }
         }
     }
 
-    protected void handlePersistentMemoryConfig(PersistentMemoryConfig persistentMemoryConfig, Node node) {
+    private void handlePersistentMemoryConfig(PersistentMemoryConfig persistentMemoryConfig, Node node) {
+        Node enabledNode = getNamedItemNode(node, "enabled");
+            if (enabledNode != null) {
+            boolean enabled = getBooleanValue(getTextContent(enabledNode));
+            persistentMemoryConfig.setEnabled(enabled);
+        }
+
+        final Node modeNode = getNamedItemNode(node, "mode");
+        final String modeStr = getTextContent(modeNode);
+        PersistentMemoryMode mode = PersistentMemoryMode.MOUNTED;
+        if (!StringUtil.isNullOrEmptyAfterTrim(modeStr)) {
+            try {
+                mode = PersistentMemoryMode.valueOf(modeStr);
+                persistentMemoryConfig.setMode(mode);
+            } catch (Exception ex) {
+                throw new InvalidConfigurationException("Invalid 'mode' for 'persistent-memory': " + modeStr);
+            }
+        }
+
         for (Node parent : childElements(node)) {
             final String nodeName = cleanNodeName(parent);
             if (matches("directories", nodeName)) {
+                if (PersistentMemoryMode.SYSTEM_MEMORY == mode) {
+                    throw new InvalidConfigurationException("Directories for 'persistent-memory' should only be"
+                            + " defined if the 'mode' is set to '" + PersistentMemoryMode.MOUNTED.name() + "'");
+                }
+
                 for (Node dirNode : childElements(parent)) {
-                    final String childNodeName = cleanNodeName(dirNode);
-                    if (matches("directory", childNodeName)) {
-                        Node numaNodeIdNode = getNamedItemNode(dirNode, "numa-node");
-                        int numaNodeId = numaNodeIdNode != null
-                                ? getIntegerValue("numa-node", getTextContent(numaNodeIdNode))
-                                : -1;
-                        String directory = getTextContent(dirNode).trim();
-                        persistentMemoryConfig.addDirectoryConfig(new PersistentMemoryDirectoryConfig(directory, numaNodeId));
-                    }
+                    handlePersistentMemoryDirectory(persistentMemoryConfig, mode, dirNode);
                 }
             }
         }
     }
 
+    protected void handlePersistentMemoryDirectory(PersistentMemoryConfig persistentMemoryConfig,
```

Comment:
Looks like this method and none of the overriden methods use the mode argument.

Reply:
Good catch, removed.